File: resources/src/main/java/com/andyanika/resources/di/module/ViewModelFactoryModule.java
package com.andyanika.resources.di.module;

import android.arch.lifecycle.ViewModelProvider;

import dagger.Binds;
import dagger.Module;

/**
 * Created by Andrey Kolpakov on 28.02.2018.
 * © Smart Space 2017
 */
@Module
public abstract class ViewModelFactoryModule {
    @Binds
    public abstract ViewModelProvider.Factory bindViewModelFactory(ViewModelFactory factory);
}

File: app/src/main/java/com/andyanika/translator/di/ViewModelFactoryModule.kt
package com.andyanika.translator.di

import androidx.lifecycle.ViewModelProvider
import dagger.Binds
import dagger.Module

/**
 * Created by Andrey Kolpakov on 28.02.2018.
 * © Smart Space 2017
 */
@Module
abstract class ViewModelFactoryModule {
    @Binds
    abstract fun bindViewModelFactory(factory: ViewModelFactory?): ViewModelProvider.Factory?
}
